ServerGUI.py

```diff
@@ -21,14 +21,17 @@ class ServerGUI:
         heading (int): integer to represent the angle that the robot is facing
     """
 
-    def __init__(self):
+    def __init__(self, init_input=None):
+        self.run_mock = init_input is not None
+
         self.master: Tk = Tk()
         self.canvas: Canvas = None
         self.tile_dict: Dict[Tile, int] = None
         self.grid = grid.Grid(tile_num_height, tile_num_width, tile_size)
         self.last_iter_seen = set()
         # TODO Update this heading in the future...
         self.heading: int = 180
+        self.desired_heading = None
```

Comment:
Why do you init to none, then check if it is none, then set it to its value?

Reply:
That is a very good question... lol